python/ray/autoscaler/autoscaler.py

```diff
@@ -312,11 +323,18 @@ def launch_config_ok(self, node_id):
         return True
 
     def files_up_to_date(self, node_id):
-        applied = self.provider.node_tags(node_id).get(TAG_RAY_RUNTIME_CONFIG)
-        if applied != self.runtime_hash:
+        node_tags = self.provider.node_tags(node_id)
+        applied_config_hash = node_tags.get(TAG_RAY_RUNTIME_CONFIG)
+        applied_file_mounts_contents_hash = node_tags.get(
+            TAG_RAY_FILE_MOUNTS_CONTENTS)
+        if (applied_config_hash != self.runtime_hash
```

Comment:
What's the reason we need to check the inner hash and not just the runtime one?

Reply:
So when the contents of the head node's file_mounts change, the files will be synced to the worker nodes.

This behavior should be the same as before since previously, the contents hash were included with the runtime hash.

Reply:
One behaviour I'm concerned with is whether "ray up" will still work if continuous_sync mode is on. If I understand correctly, that won't update the cluster if only files have changed right?

Reply:
ah I see, in that case the rsync will still happen but the setup commands wont run. I'll make a change so that it only doesn't run setup commands when update is called in the monitor.py flow.